src/firefly/js/fieldGroup/FieldGroupUtils.js

```diff
@@ -5,84 +5,96 @@
 import {get} from 'lodash';
 import {flux} from '../Firefly.js';
 import {logError} from '../util/WebUtil.js';
-import FieldGroupCntlr from './FieldGroupCntlr.js';
+import FieldGroupCntlr, {dispatchValueChange} from './FieldGroupCntlr.js';
+
+
+const includeForValidation= (f,includeUnmounted) => f.valid !== undefined && (f.mounted||includeUnmounted);
+
+
+function validateResolvedSingle(groupKey,includeUnmounted, doneCallback) {
+    var flds= getGroupFields(groupKey);
+    var valid = Object.keys(flds).every( (key) => includeForValidation(flds[key]) ? flds[key].valid : true);
+    doneCallback(valid);
+}
 
-/**
- * make a promise for this field key to guarantee that all async validation has completed
- * @param fields the fields
- * @param fieldKey the field key to convert to non async
- * @return Promise
- */
-var makeValidationPromise= function(fields,fieldKey) {
-    if (fields[fieldKey].mounted && fields[fieldKey].asyncUpdatePromise) {
-        return fields[fieldKey].asyncUpdatePromise;
-    }
-    else {
-        return Promise.resolve(fieldKey);
-    }
-};
 
 /**
- *
- * @return Promise with the valid state true/false
+ * 
+ * @param groupKey
+ * @param includeUnmounted
+ * @param doneCallback
+ * @return Promise with the valid state true/false --- todo: the return value does not work
  */
-var validateSingle= function(groupKey, doneCallback) {
+var validateSingle= function(groupKey, includeUnmounted, doneCallback) {
     var fields= getGroupFields(groupKey);
+
+    //====== clear out all functions
+    Object.keys(fields).forEach( (key) => {
+        if (typeof fields[key].value === 'function') {
+            var newValue= fields[key].value();
+            if (typeof newValue=== 'object' &&       // check to see if return is an object with {value:string,valid:boolean}
+                Object.keys(newValue).length===2 &&
+                newValue.hasOwnProperty('valid') &&
+                newValue.hasOwnProperty('value') ) {
+                dispatchValueChange({fieldKey:key,groupKey,valid:newValue.valid,value:newValue.value});
+            }
+            else {
+                dispatchValueChange({fieldKey:key,groupKey,valid:true,value:newValue});
+            }
+        }
```

Comment:
The field value would be a function on the file upload case. Therefore the upload does not activate until validation.  In the upload case the function would return a promise.  However, It could return a value or an object with a value and a valid status,